demos/companion-app/src/components/fund-arcade-account.tsx

```diff
@@ -0,0 +1,137 @@
+"use client";
+import {
+  useAccount,
+  useContractWrite,
+  useContractRead,
+} from "@starknet-react/core";
+import { useMemo, useState } from "react";
+import { CallData, stark, uint256 } from "starknet";
+import { z } from "zod";
+import { abi } from "@/lib/factory";
+import { Input } from "./ui/input";
+import { useForm } from "react-hook-form";
+import { Label } from "./ui/label";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "./ui/button";
+import { Card, CardContent } from "./ui/card";
+import { bnToHex } from "@/lib/utils";
+import { contractAddress } from "@/lib/constants";
+
+interface Props {
+  pk: string;
+}
+
+const schema = z.object({
+  initialSupply: z.number().min(1),
+});
+
+export const FundArcadeAccount = ({ pk }: Props) => {
+  const { address, account } = useAccount();
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+    reset: resetForm,
+  } = useForm<z.infer<typeof schema>>({
+    resolver: zodResolver(schema),
+    defaultValues: { initialSupply: 10_000_000_000 },
+  });
+
+  const salt = useMemo(() => {
+    return stark.randomAddress();
+  }, [account]);
+
+  const { writeAsync, reset: resetWrite, error } = useContractWrite({});
+
+  const [deployedToken, setDeployedToken] = useState<
+    { address: string; tx: string } | undefined
+  >(undefined);
+
+  const amount = watch("initialSupply");
+
+  const { data: tokenAddress } = useContractRead({
+    abi,
+    address: contractAddress,
+    functionName: "compute_address",
+    args: [salt, pk, address as string],
+  });
+
+  const deployToken = async (data: z.infer<typeof schema>) => {
+    if (!address) return;
+
+    const hexAddres = bnToHex(tokenAddress as bigint);
+
+    const deploy = {
+      contractAddress: contractAddress,
+      entrypoint: "deploy",
+      calldata: [salt, pk, address],
+    };
+
+    const transfer = {
+      contractAddress: hexAddres,
```

Comment:
Here we want to transfer the native token (ETH).
```suggestion
      contractAddress: chain.nativeCurrency.address,
```